File: problem-6/sumsqrdiff.hs
sumsqrto :: Int -> Int -> Int
sumsqrto n max
    | n >  max  = 0
    | otherwise = (n * n) + sumsqrto (n + 1) max

sumto :: Int -> Int -> Int
sumto n  max
    | n > max   = 0
    | otherwise = n + sumto (n + 1) max

main = do
    print( (sumto 0 100) ^ 2 - sumsqrto 0 100)

File: problem-006/elixir/sumsqrdiff.exs
defmodule P006 do
  def sumsqrto(n, max) do
    cond do
      n > max -> 0
      true -> (n * n) + sumsqrto((n + 1), max)
    end
  end

  def sumto(n, max) do
    cond do
      n > max -> 0
      true -> n + sumto((n + 1), max)
    end
  end
end

IO.puts((P006.sumto(0, 100))**2 - P006.sumsqrto(0, 100))
